articles/native-platforms/xamarin/02-custom-login.md

```diff
@@ -0,0 +1,269 @@
+			public string picture { get; set; }
+			public string email { get; set; }
+		}
+```
+
+As you can see the REST API method requires a lot of additional code, but is a valid way to authenticate if needed.
+
+## Social Login
+
+The Auth0 SDK also allows you to implement social login with just a few lines of code. As with the custom login the first step is to instantiate the Auth0Client:
+
+${snippet(meta.snippets.setup)}
+
+Next, let's build the user interface we wish to present to the user.
+
+```cs
+var facebookLoginButton = new Button{ Text = "Login with Facebook" };
+
+facebookLoginButton.Clicked += (object sender, EventArgs e) =>{
+  LoginWithFacebook();
+};
+
+Content = new StackLayout{
+  Padding = 30,
+  Spacing = 10,
+  Children = {
+    new Label { Text = "Social Login" },
+    facebookLoginButton
+  }
+};
+```
+
+In addition to the Auth0 SDK we will also require a couple of additional libraries. Since we want to make sure the social login works on both Android and iOS devices, we'll include the required libraries to handle the WebView display. At the top of your file be sure to include the following:
+
+```cs
+#if __IOS__
+  using UIKit;
+#endif
+#if __ANDROID__
+  using Android.Content;
+#endif
+```
+
+Next, implement the `LoginWithFacebook()` function which will make use of the Auth0 SDK to handle the authentication.
+
+```cs
+public async void LoginWithFacebook()
+{
+  #if __IOS__
+    var window = UIApplication.SharedApplication.KeyWindow;
+    var ui = window.RootViewController;
+    // This will guarantee that the Facebook login is displayed above all other screens.
+    while (ui.PresentedViewController != null)
+    {
+      ui = ui.PresentedViewController;
+    }
+  #endif
+  #if __ANDROID__
+    Context ui = Xamarin.Forms.Forms.Context;
+  #endif
+
+  // The second paramater here is the social network. In our example we are using Facebook, but any supported IdP would work here.
+  var user = await auth0.LoginAsync(ui, "facebook");
+
+  Application.Current.Properties["id_token"] = user.IdToken;
+  Application.Current.Properties["access_token"] = user.Auth0AccessToken;
+
+  Application.Current.Properties["name"] = user.Profile["name"].ToString();
+  Application.Current.Properties["email"] = user.Profile["email"].ToString();
+  Application.Current.Properties["picture"] = user.Profile["picture"].ToString();
+
+  await Navigation.PushModalAsync(new HomePage());
+}
+```
+
+The `Auth0User` has the following properties:
+
+* `Profile`: returns a `Newtonsoft.Json.Linq.JObject` object from [Json.NET component](http://components.xamarin.com/view/json.net/) containing all available user attributes (e.g.:`user.Profile["email"].ToString()`).
+* `IdToken`: a JSON Web Token (JWT) containing all of the user attributes and signed with your client secret.
+* `Auth0AccessToken`: the `access_token` that can be used to call the Auth0 APIs. For example, you could use this token to [Link Accounts](/link-accounts).
```

Comment:
I'd remove the mention of link accounts and state that the `access_token` can be used with an API if the OAuth 2.0 features are enabled (and supplied an audience during Auth).